Sources/SwiftSyntaxMacros/Syntax+LexicalContext.swift

```diff
@@ -67,6 +67,20 @@ extension SyntaxProtocol {
     case let freestandingMacro as FreestandingMacroExpansionSyntax:
       return Syntax(freestandingMacro.detached) as Syntax
 
+    // Try and await are preserved: A freestanding expression macro preceded
+    // by try or await may need to know whether those keywords are present so it
+    // can propagate them to any expressions in its expansion which were passed
+    // as arguments to the macro. The expression of the try or await is replaced
+    // with a trivial placeholder, though.
+    case var tryExpr as TryExprSyntax:
+      tryExpr = tryExpr.detached
+      tryExpr.expression = ExprSyntax(TypeExprSyntax(type: IdentifierTypeSyntax(name: .wildcardToken())))
```

Comment:
The wildcard token was chosen arbitrarily here. We don't necessarily have to reassign the expression at all, but I figured it shouldn't be necessary since the macro implementation already has it.

Reply:
Arguably we _do_ need to reassign it, otherwise in an expression like this one:

```swift
await f(a, b, c, #d)
```

`d` is able to see `a`, `b`, and `c`.

Reply:
A syntax tree that hasn't had operator folding applied won't have the expected `try` or `await` parent expressions. For example, something like

```swift
func f() {
  let x = try a + #require(b)
}
```

won't have the `#require(b)` nested under the `try`.